File: debug_lcp.py
import json
import sys

try:
    with open(sys.argv[1]) as f:
        data = json.load(f)
    
    lcp_audit = data['audits']['largest-contentful-paint']
    fcp_audit = data['audits']['first-contentful-paint']
    
    print(f"FCP Numeric: {fcp_audit.get('numericValue')}")
    print(f"LCP Numeric: {lcp_audit.get('numericValue')}")
    
    # Look for the element audit specifically
    if 'audits' in data and 'largest-contentful-paint-element' in data['audits']:
        lcp_element_audit = data['audits']['largest-contentful-paint-element']
        if 'details' in lcp_element_audit and 'items' in lcp_element_audit['details']:
            print("LCP Element Details:")
            print(json.dumps(lcp_element_audit['details']['items'], indent=2))
        else:
             print("LCP Element audit found but no items.")
    else:
        print("largest-contentful-paint-element audit not found")

except Exception as e:
    print(f"Error: {e}")

File: scripts/rust/src/bin/debug_lcp.rs
//! Debug LCP (Largest Contentful Paint) from Lighthouse JSON
//! Rust replacement for debug_lcp.py

use anyhow::{Context, Result};
use serde_json::Value;
use std::env;
use std::fs;

fn main() -> Result<()> {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Usage: debug_lcp <lighthouse-report.json>");
        std::process::exit(1);
    }

    let file_path = &args[1];
    let content = fs::read_to_string(file_path)
        .with_context(|| format!("Failed to read file: {}", file_path))?;

    let data: Value = serde_json::from_str(&content)
        .context("Failed to parse JSON")?;

    // Extract FCP
    let fcp_numeric = data["audits"]["first-contentful-paint"]["numericValue"]
        .as_f64()
        .unwrap_or(0.0);
    println!("FCP Numeric: {:.2}", fcp_numeric);

    // Extract LCP
    let lcp_numeric = data["audits"]["largest-contentful-paint"]["numericValue"]
        .as_f64()
        .unwrap_or(0.0);
    println!("LCP Numeric: {:.2}", lcp_numeric);

    // Extract LCP Element Details
    if let Some(lcp_element_audit) = data["audits"].get("largest-contentful-paint-element") {
        if let Some(items) = lcp_element_audit["details"]["items"].as_array() {
            println!("LCP Element Details:");
            println!("{}", serde_json::to_string_pretty(items)?);
        } else {
            println!("LCP Element audit found but no items.");
        }
    } else {
        println!("largest-contentful-paint-element audit not found");
    }

    Ok(())
}
